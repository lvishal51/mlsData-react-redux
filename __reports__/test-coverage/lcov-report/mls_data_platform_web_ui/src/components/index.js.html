<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for mls_data_platform_web_ui/src/components/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">mls_data_platform_web_ui/src/components</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export { default as ZapInput } from './zap-input/ZapInput';
export { default as ZapHeader } from './zap-header/ZapHeader';
export { default as ZapFooter } from './zap-footer/ZapFooter';
export { default as ZapPanel } from './zap-panel/ZapPanel';
export { default as MappedUnmapped } from './zap-mapping/MappedUnmapped';
export { default as CustomPagination } from './zap-pagination/CustomPagination';
export { default as MappedFields } from './zap-mappedFields/MappedFields';
export { default as ContextDataPanel } from './zap-context-panel/ContextDataPanel';
export { default as AutoSuggestion } from './zap-autoSuggestion/AutoSuggestion';
export { default as ZapTable } from './zap-table/ZapTable';
export { default as AutoComplete } from './zap-autoComplete/AutoComplete';
export { default as ZapSideBar } from './zap-sideBar/zapSideBar';
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 29 2018 15:48:26 GMT+0530 (IST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
