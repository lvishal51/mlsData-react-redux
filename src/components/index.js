export { default as ZapInput } from './zap-input/ZapInput';
export { default as ZapHeader } from './zap-header/ZapHeader';
export { default as ZapFooter } from './zap-footer/ZapFooter';
export { default as ZapPanel } from './zap-panel/ZapPanel';
export { default as MappedUnmapped } from './zap-mapping/MappedUnmapped';
export { default as CustomPagination } from './zap-pagination/CustomPagination';
export { default as MappedFields } from './zap-mappedFields/MappedFields';
export { default as ContextDataPanel } from './zap-context-panel/ContextDataPanel';
export { default as AutoSuggestion } from './zap-autoSuggestion/AutoSuggestion';
export { default as ZapPreview } from './zap-preview/ZapPreview';
export { default as AutoComplete } from './zap-autoComplete/AutoComplete';
export { default as CustomCalendar } from './zap-calendar/CustomCalendar';
export { ZapHeader as ZapHeaderComp } from './zap-header/ZapHeader';
export { ZapPanel as ZapPanelComp } from './zap-panel/ZapPanel';
export { ContextDataPanel as ContextDataPanelComp} from './zap-context-panel/ContextDataPanel';
export { AutoComplete as AutoCompleteComp } from './zap-autoComplete/AutoComplete';
export { MappedFields as MappedFieldsComp} from './zap-mappedFields/MappedFields';
