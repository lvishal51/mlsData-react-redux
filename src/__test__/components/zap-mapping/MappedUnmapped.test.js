import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { MappedUnmapped } from 'components';

const mlsStandardValueMapDtos = [{
  changeDate: 1548786600000,
  createDate: 1548786600000,
  enteredBy: 0,
  fieldType: "AGE",
  id: 5096,
  source: "FLEXMLS",
  standardValueId: 259079,
  value: "Ag106"
}];

const mappedData = {
  payload : [
  {
    "id": 25,
    "value": "CURRENT_TAXES",
    "type": "REPORT",
    "dataType": "MEDIUMTEXT",
    "dataLength": 0,
    "notes": null,
    "lastUpdateTs": "2018-10-15T18:30:00.000Z",
    "createTs": null,
    mlsStandardValueMapDtos
  }
]};

const selectedMapped = [
  {
    "id": 25,
    "value": "CURRENT_TAXES",
    "type": "REPORT",
    "dataType": "MEDIUMTEXT",
    "dataLength": 0,
    "notes": null,
    "lastUpdateTs": "2018-10-15T18:30:00.000Z",
    "createTs": null
  }
];

const mlsUnmappedValueDtos = [{
  changeDate: null,
  createDate: null,
  fieldType: "AGE",
  id: null,
  source: "FLEXMLS",
  value: "ADDRESS"
}];

const unMappedData ={
  payload :
  {
    "id": 1,
    "value": "ADDRESS",
    "type": "REPORT",
    "dataType": "MEDIUMTEXT",
    "dataLength": 0,
    "notes": null,
    "lastUpdateTs": "2018-10-15T18:30:00.000Z",
    "createTs": null,
    mlsUnmappedValueDtos
  }
 };

const selectedUnmapped = [
  {
    "id": 1,
    "value": "ADDRESS",
    "type": "REPORT",
    "dataType": "MEDIUMTEXT",
    "dataLength": 0,
    "notes": null,
    "lastUpdateTs": "2018-10-15T18:30:00.000Z",
    "createTs": null
  }
]

const inputData = {
    mappedData,
    unMappedData,
    moveToUnmapped: () => {},
    moveToMapped: () => {}
}

describe('Zap Mapped Unmapped component', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<MappedUnmapped {...inputData}/>);
  });

  it('should render correctly.', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('onSearch function should be triggered correctly in case mapped', () => {
    const event = { preventDefault() {}, target: { value: 'ADDRESS' } };
    wrapper.setProps({ onSearchMappedValues: () => {}, onSearchUnMappedValues: () => {}});
    const spy = jest.spyOn(wrapper.instance(), "onSearch");
    wrapper.instance().forceUpdate();
    wrapper.find(".mappedSearch").simulate('change', event, 'mapped');
    expect(spy).toBeCalled();
  });

  it('selectContent function remove selection unmappedCase', () => {
    wrapper.setState({selectedUnmapped});
    const event = { preventDefault() {}, target: { value: 'ADDRESS' } };
    const spy = jest.spyOn(wrapper.instance(), "selectContent");
    wrapper.instance().forceUpdate();
    wrapper.find(".unmapped").simulate('click', event, 'unmapped');
    expect(spy).toBeCalled();
  });

  it('selectContent function in case no mtach found', () => {
    wrapper.setState({selectedUnmapped: selectedMapped});
    const event = { preventDefault() {}, target: { value: 'ADDRESS' } };
    const spy = jest.spyOn(wrapper.instance(), "selectContent");
    wrapper.instance().forceUpdate();
    wrapper.find(".unmapped").simulate('click', event, 'unmapped');
    expect(spy).toBeCalled();
  });

  it('selectContent function should be triggered correctly in case unmapped', () => {
    const event = { preventDefault() {}, target: { value: 'ADDRESS' } };
    const spy = jest.spyOn(wrapper.instance(), "selectContent");
    wrapper.instance().forceUpdate();
    wrapper.find(".unmapped").simulate('click', event, 'unmapped');
    expect(spy).toBeCalled();
  });

  it('moveToMapped function should be triggered correctly', () => {
    wrapper.setState({selectedUnmapped});
    const spy = jest.spyOn(wrapper.instance(), "moveToMapped");
    wrapper.instance().forceUpdate();
    wrapper.find(".mappedArrow").simulate('click');
    expect(spy).toBeCalled();
  });

  it('moveToMapped function when nothing is selected', () => {
    wrapper.setState({selectedUnmapped: []});
    const spy = jest.spyOn(wrapper.instance(), "moveToMapped");
    wrapper.instance().forceUpdate();
    wrapper.find(".mappedArrow").simulate('click');
    expect(spy).toBeCalled();
  });

  it('moveToUnmapped function should be triggered correctly', () => {
    wrapper.setState({selectedMapped});
    const spy = jest.spyOn(wrapper.instance(), "moveToUnmapped");
    wrapper.instance().forceUpdate();
    wrapper.find(".unmappedArrow").simulate('click');
    expect(spy).toBeCalled();
  });

  it('moveToUnmapped function when nothing is selected', () => {
    wrapper.setState({selectedMapped: []});
    const spy = jest.spyOn(wrapper.instance(), "moveToUnmapped");
    wrapper.instance().forceUpdate();
    wrapper.find(".unmappedArrow").simulate('click');
    expect(spy).toBeCalled();
  });

  it('onNextClick function for mappedCase', () => {
    wrapper.setState({selectedMapped});
    wrapper.setProps({ onUnMappedNextClick: () => {}});
    const spy = jest.spyOn(wrapper.instance(), "onNextClick");
    wrapper.instance().forceUpdate();
    wrapper.find(".nextLink").simulate('click', 'mapped');
    expect(spy).toBeCalled();
  });

  it('onPreviousClick function for mappedCase', () => {
    wrapper.setState({selectedMapped});
    wrapper.setProps({ onUnMappedPreviousClick: () => {}});
    const spy = jest.spyOn(wrapper.instance(), "onPreviousClick");
    wrapper.instance().forceUpdate();
    wrapper.find(".previousLink").simulate('click', 'mapped');
    expect(spy).toBeCalled();
  });
});
