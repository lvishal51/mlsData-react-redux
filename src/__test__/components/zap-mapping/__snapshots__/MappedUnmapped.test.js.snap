// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zap Mapped Unmapped component should render correctly. 1`] = `
<Row
  className="mappingContainer"
  tag="div"
>
  <Col
    lg="5"
    md="5"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="12"
  >
    <Card
      tag="div"
    >
      <CardHeader
        className="h5"
        tag="div"
      >
        Unmapped Values
      </CardHeader>
      <CardBody
        tag="div"
      >
        <FormGroup
          tag="div"
        >
          <DebounceInput
            className="form-control unmappedSearch"
            debounceTimeout={300}
            element="input"
            forceNotifyByEnter={true}
            forceNotifyOnBlur={true}
            minLength={0}
            name="search"
            onChange={[Function]}
            placeholder="Search Unmapped Values"
            type="search"
          />
        </FormGroup>
        <Row
          className="mb-3 mt-3"
          tag="div"
        >
          <Col
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
            xs="6"
          >
            <a
              className="disabledNavigationLinks previousLink"
              href="javascript:void(0)"
              onClick={[Function]}
            >
              Previous
            </a>
          </Col>
          <Col
            className="text-right"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
            xs="6"
          >
            <a
              className="nextLink"
              href="javascript:void(0)"
              onClick={[Function]}
            >
              Next
            </a>
          </Col>
        </Row>
        <Row
          className="listScroll"
          tag="div"
        >
          <Col
            lg="12"
            md="12"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
            xs="12"
          >
            <ListGroup
              flush={true}
              tag="ul"
            >
              <ListGroupItem
                className="unmapped"
                key="0"
                onClick={[Function]}
                tag="li"
              >
                ADDRESS
              </ListGroupItem>
            </ListGroup>
          </Col>
        </Row>
      </CardBody>
    </Card>
  </Col>
  <Col
    className="moveItemWithArrow"
    lg="2"
    md="2"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="12"
  >
    <div
      className="arrowsSection"
    >
      <Button
        className="fa fa-chevron-circle-right custom mappedArrow zapButton"
        color="secondary"
        disabled={true}
        onClick={[Function]}
        tag="button"
      />
      <Button
        className="fa fa-chevron-circle-left custom unmappedArrow zapButton"
        color="secondary"
        disabled={true}
        onClick={[Function]}
        tag="button"
      />
    </div>
  </Col>
  <Col
    lg="5"
    md="5"
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
    xs="12"
  >
    <Card
      tag="div"
    >
      <CardHeader
        className="h5"
        tag="div"
      >
        Mapped Values
      </CardHeader>
      <CardBody
        tag="div"
      >
        <FormGroup
          tag="div"
        >
          <DebounceInput
            className="form-control mappedSearch"
            debounceTimeout={300}
            element="input"
            forceNotifyByEnter={true}
            forceNotifyOnBlur={true}
            minLength={0}
            name="search"
            onChange={[Function]}
            placeholder="Search Mapped Values"
            type="search"
          />
        </FormGroup>
        <Row
          className="listScroll"
          tag="div"
        >
          <Col
            lg="12"
            md="12"
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
            xs="12"
          >
            <ListGroup
              flush={true}
              tag="ul"
            />
          </Col>
        </Row>
      </CardBody>
    </Card>
  </Col>
</Row>
`;
